Src/VimWpf/Implementation/CommandMargin/CommandMarginControl.xaml

```diff
@@ -43,45 +43,42 @@
             CaretBrush="{Binding Path=TextForeground}"
             />
 
-        <ScrollViewer 
-            Grid.Row="1" 
-            Grid.Column="1"
-            HorizontalScrollBarVisibility="Auto"
-            VerticalScrollBarVisibility="Disabled">
-            <Border BorderThickness="1" BorderBrush="{Binding ElementName=_commandLineInput, Path=BorderBrush}">
-                <Grid>
-                    <Grid.ColumnDefinitions>
-                        <ColumnDefinition Width="*" />
-                        <ColumnDefinition Width="Auto" />
-                    </Grid.ColumnDefinitions>
-                    <TextBox
-                        x:Name="_commandLineInput"
-                        MaxHeight="200"
-                        BorderThickness="0"
-                        VerticalAlignment="Center"
-                        IsReadOnly="{Binding Path=IsEditReadOnly}"
-                        FontFamily="{Binding Path=TextFontFamily}"
-                        FontSize="{Binding Path=TextFontSize}"
-                        Foreground="{Binding Path=TextForeground}"
-                        Background="{Binding Path=TextBackground}"
-                        CaretBrush="{Binding Path=TextForeground}" 
-                        />
-                    <TextBox
-                        Grid.Row="0"
-                        Grid.Column="1"
-                        MinWidth="50"
-                        IsReadOnly="True"
-                        x:Name="_showCommandText"
-                        BorderThickness="0"
-                        VerticalAlignment="Center"
-                        FontFamily="{Binding Path=TextFontFamily}"
-                        FontSize="{Binding Path=TextFontSize}"
-                        Foreground="{Binding Path=TextForeground}"
-                        Background="{Binding Path=TextBackground}"
-                        CaretBrush="{Binding Path=TextForeground}" 
-                        />
-                </Grid>
-            </Border>
-        </ScrollViewer>

```

Comment:
Do you understand why the `ScrollViewer` didn't work in the old XAML code? Asking because I am curious what was wrong here.

Reply:
I think my choice to move the scrolling to the text box (which has a scrollviewer embedded) has obscured the underlying issue.  I'll try to explain both.

The reason the scroll viewer wasn't showing its vertical scrollbar (even after being set to "Auto") was that its height was unconstrained.  An unconstrained scroll viewer will happily resize itself to accommodate its contents.  Since scroll bars are only enabled when the scroll viewer's content area is smaller than its contents, you would never have seen the vertical scrollbar.

Moving `MaxHeight="200"` from `_commandLineInput` to the `ScrollViewer` tag would have fixed this particular problem. On the other hand, the scrollbar is now **outside** of the text box and the thin border around it.  If the text box were tall enough to cause the vertical scrollbar to show, then the bottom border would have disappeared "off screen", which I thought looked weird (see image below).  The simplest solution was to hand over scrolling to the scroll viewer embedded within the text box itself.

![image](https://user-images.githubusercontent.com/3457284/44266583-0f466380-a1f9-11e8-9268-28cb4b8665c9.png)

Reply:
To contrast with the above image, here's the scroll bar **inside** the text box:
![image](https://user-images.githubusercontent.com/3457284/44266982-8af4e000-a1fa-11e8-8239-2980cdc0d672.png)

Reply:
Thanks for the explanation!